src/PowerShellEditorServices/Console/IConsoleOperations.cs

```diff
@@ -0,0 +1,22 @@
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.PowerShell.EditorServices.Console
+{
+    /// <summary>
+    /// Provides platform specific console utilities.
+    /// </summary>
+    public interface IConsoleOperations
+    {
+        /// <summary>
+        /// Obtains the next character or function key pressed by the user asynchronously.
+        /// Does not block when other console API's are called.
+        /// </summary>
+        /// <param name="cancellationToken">The CancellationToken to observe.</param>
+        /// <returns>
+        /// A task that will complete with a result of the key pressed by the user.
+        /// </returns>
+        Task<ConsoleKeyInfo> ReadKeyAsync(CancellationToken cancellationToken);
+    }
+}
```

Comment:
For POSIX  compliance, don't we want a newline at the end of the file?  See https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline

Reply:
POSIX compliance is sometimes overrated :)

Reply:
I was just thinking if someone wanted to work on this repo on Linux/macOS.  OTOH they'd probably be using PSCore utilities on files and not the native utilities.

Reply:
@rkeithhill good catch, I usually have the VSCode workspace setting that adds it automatically.  Any objections to me adding it here?

Reply:
Not from me.  In fact, add that setting to the .vscode\settings.json file.  Or maybe we adopt .editorconfig so that VS 2017 will honor this setting.  There's a plugin for VSCode for EditorConfig so that VSCode will also honor the setting.

Reply:
@rkeithhill added to the `settings.json` for now, but I agree with translating that to an editor config.  I'll make a separate issue for that.